File: Golang/496. Next Greater Element I.go
func nextGreaterElement(findNums []int, nums []int) []int {
    ans := make([]int, 0, len(findNums))
    length := 0
    for _, num1 := range findNums {
        length++
        index := 0
        for i, num2 := range nums {
            if num1 == num2 {
                index = i
                break
            }
        }
        for _, num2 := range nums[index:] {
            if num1 < num2 {
                ans = append(ans, num2)
                break
            }
        }
        if len(ans) < length {
            ans = append(ans, -1)
        }
    }
    return ans
}

File: Python3/496. Next Greater Element I.py
class Solution:
    def nextGreaterElement(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        ans = []
        length = 0
        for num in nums1:
            length += 1
            index = nums2.index(num)
            for num2 in nums2[index:]:
                if num2 > num:
                    ans.append(num2)
                    break
            if len(ans) < length:
                ans.append(-1)
        return ans
